lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/HeartBeatManager.cs

```diff
@@ -126,6 +128,18 @@ public void Send(EvaluatorHeartbeatProto evaluatorHeartbeatProto)
         {
             lock (_queuedHeartbeats)
             {
+                // Do not send a heartbeat if Evaluator has already signaled that it was done.
+                if (_isCompletedHeartbeatQueued)
```

Comment:
No, `evaluatorHeartbeatProto` is a local variable that differs across heartbeats. We need something to keep the state across different heartbeats, and we cannot do it without introducing a `private` instance variable.